operations/src/test/java/edu/unc/lib/boxc/operations/impl/download/DownloadBulkServiceTest.java

```diff
@@ -0,0 +1,231 @@
+                    anyString(), eq(parentPid), any(), eq(Permission.viewOriginal));
+
+            service.downloadBulk(request);
+        });
+    }
+
+    @Test
+    public void notAWorkTest() {
+        when(repoObjLoader.getWorkObject(any(PID.class))).thenReturn(null);
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            service.downloadBulk(request);
+        });
+    }
+
+    @Test
+    public void workObjectDoesNotHaveFileObjectsTest() throws IOException {
+        when(repoObjLoader.getWorkObject(any(PID.class))).thenReturn(parentWork);
+        service.downloadBulk(request);
+        // the zip file should be empty
+        assertZipFile(0, List.of());
+    }
+
+    @Test
+    public void noAccessToFileObjectsTest() throws IOException {
+        when(repoObjLoader.getWorkObject(any(PID.class))).thenReturn(parentWork);
+        when(parentWork.getMembers()).thenReturn(List.of(fileObject1));
+        when(aclService.hasAccess(eq(fileObject1Pid), any(),
+                eq(Permission.viewOriginal))).thenReturn(false);
+        service.downloadBulk(request);
+        // the zip file should be empty
+        assertZipFile( 0, List.of());
+    }
+
+    @Test
+    public void accessToOnlyOneFileObjectTest() throws IOException {
+        when(repoObjLoader.getWorkObject(any(PID.class))).thenReturn(parentWork);
+        when(parentWork.getMembers()).thenReturn(List.of(fileObject1, fileObject2));
+        makeBinaryObject(fileObject1, FILENAME1);
+        when(aclService.hasAccess(eq(fileObject1Pid), any(),
+                eq(Permission.viewOriginal))).thenReturn(true);
+        when(aclService.hasAccess(eq(fileObject2Pid), any(),
+                eq(Permission.viewOriginal))).thenReturn(false);
+        service.downloadBulk(request);
+        // the zip file should have one entry
+        assertZipFile(1, List.of(FILENAME1));
+    }
+
+    @Test
+    public void noOriginalFilesTest() throws IOException {
+        when(repoObjLoader.getWorkObject(eq(parentPid))).thenReturn(parentWork);
+        when(parentWork.getMembers()).thenReturn(List.of(fileObject1));
+        service.downloadBulk(request);
+        // the zip file should be empty
+        assertZipFile(0, List.of());
+    }
+
+    @Test
+    public void successTest() throws IOException {
+        when(repoObjLoader.getWorkObject(eq(parentPid))).thenReturn(parentWork);
+        when(parentWork.getMembers()).thenReturn(List.of(fileObject1, fileObject2));
+        when(aclService.hasAccess(eq(fileObject1Pid), any(),
+                eq(Permission.viewOriginal))).thenReturn(true);
+        when(aclService.hasAccess(eq(fileObject2Pid), any(),
+                eq(Permission.viewOriginal))).thenReturn(true);
+        makeBinaryObject(fileObject1, FILENAME1);
+        makeBinaryObject(fileObject2, FILENAME2);
+
+        service.downloadBulk(request);
+
+        // the zip file should have two entries
+        assertZipFile(2, List.of(FILENAME1, FILENAME2));
```

Comment:
I'm wondering if it'd be good to verify at least one time that the file contained in the zip has the content "flower", just to ensure we aren't getting empty files copied in or something